lib/slither/generator.rb

```diff
@@ -10,8 +10,8 @@ def generate(data)
 	    @definition.sections.each do |section|
 	      content = data[section.name]
 	      if content
-  	      content = [content] unless content.is_a?(Array)
-  	      raise(Slither::RequiredSectionEmptyError, "Required section '#{section.name}' was empty.") if content.empty?
+  	      content = [content] if content.is_a?(Hash)
+  	      raise(Slither::RequiredSectionEmptyError, "Required section '#{section.name}' was empty.") unless content.any?
```

Comment:
I also notice the original spec omitted the test case for "when section exists with no content". Since you are updating this check, let's add in the test case.